public/components/event_analytics/explorer/events_views/data_grid.tsx

```diff
@@ -3,148 +3,271 @@
+      const columns: EuiDataGridColumn[] = [];
+      storedSelectedColumns.map(({ name, type }) => {
+        if (name === 'timestamp') {
+          columns.push(DEFAULT_TIMESTAMP_COLUMN);
+        } else if (name === '_source') {
+          columns.push(DEFAULT_SOURCE_COLUMN);
+        } else {
+          columns.push({
+            id: name,
+            display: name,
+            isSortable: true, // TODO: add functionality here based on type
+          });
+        }
+      });
+      return columns;
+    }
+    return [];
+  }, [storedSelectedColumns]);
+
+  // used for which columns are visible and their order
+  const dataGridColumnVisibility = useMemo(() => {
+    if (storedSelectedColumns.length > 0) {
+      const columns: string[] = [];
+      storedSelectedColumns.map(({ name }) => {
+        columns.push(name);
+      });
+      return {
+        visibleColumns: columns,
+        setVisibleColumns: (visibleColumns: string[]) => {
+          // TODO: implement with sidebar field order (dragability) changes
+        },
+      };
+    }
+    // default shown fields
+    return {
+      visibleColumns: [],
+      setVisibleColumns: () => {},
+    };
+  }, [storedSelectedColumns]);
+
+  // sets the very first column, which is the button used for the flyout of each row
+  const dataGridLeadingColumns = useMemo(() => {
+    return [
+      {
+        id: 'inspectCollapseColumn',
+        headerCellRender: () => null,
+        rowCellRender: ({ rowIndex }: { rowIndex: number }) => {
+          return (
+            <FlyoutButton
+              ref={null}
+              http={http}
+              key={null}
+              docId={'undefined'}
+              doc={rows[rowIndex % pageFields.current[1]]}
+              selectedCols={explorerFields.queriedFields}
+              timeStampField={timeStampField}
+              explorerFields={explorerFields}
+              pplService={pplService}
+              rawQuery={rawQuery}
+              onFlyoutOpen={onFlyoutOpen}
+            />
+          );
+        },
+        width: 40,
+      },
+    ];
+  }, [rows, http, explorerFields, pplService, rawQuery, onFlyoutOpen]);
+
+  // renders what is shown in each cell, i.e. the content of each row
+  const dataGridCellRender = useCallback(
+    ({ rowIndex, columnId }: { rowIndex: number; columnId: string }) => {
+      const trueIndex = rowIndex % pageFields.current[1];
+      if (trueIndex < rows.length) {
+        if (columnId === '_source') {
+          return (
+            <EuiDescriptionList type="inline" compressed>
+              {Object.keys(rows[trueIndex]).map((key) => (
+                <Fragment key={key}>
+                  <EuiDescriptionListTitle className="osdDescriptionListFieldTitle">
+                    {key}
+                  </EuiDescriptionListTitle>
+                  <EuiDescriptionListDescription
+                    dangerouslySetInnerHTML={{ __html: dompurify.sanitize(rows[trueIndex][key]) }}
```

Comment:
why is this needed rather than passing as children?

Reply:
I assume the thinking was to sanitize the returned data before displaying to prevent XSS but I don't think it's needed here. I changed it back to passing the row value as a child

Reply:
sanitize is needed for `dangerouslySetInnerHTML`, i'm asking is it necessary to `dangerouslySetInnerHTML` in the first place? if not probably just pass as child

Reply:
done, i'm passing it in as a child.